demo/Titanic/simulation/titanic_in_ocean.py

```diff
@@ -1,17 +1,41 @@
+from __future__ import annotations
+
 from simulation.floating_object import FloatingObject
-from system.titanic import Titanic
 
 
 class TitanicInOcean(FloatingObject):
+
+    instance: TitanicInOcean = None
+
     def __new__(cls, titanic, n, w, speed, direction, sunk=False):
-        if not hasattr(cls, "instance"):
+        if not cls.instance:
             cls.instance = super(TitanicInOcean, cls).__new__(cls)
         return cls.instance
 
-    def __init__(self, titanic, n, w, speed, direction, sunk=False):
+    def __init__(self, titanic, n, w, speed, direction):
         super().__init__(n, w, speed, direction)
-        self.sunk = sunk
+        self.sunk = False
         self.titanic = titanic
 
+    def update_coordinates(self):
+        # TODO Take into account the steering_direction of the titanic
+        # TODO Take into account the throttle of the titanic (i.e.: should start decreasing speed when throttle is 0)
+        return super().update_coordinates()
```

Comment:
Test run fails here due to missing argument definition compared to the base class. I think the method can be omitted in full.

Reply:
It can be, but i wanted to keep the todos in there such that they can be picked up. Fixed now

Reply:
Good point, we will need that.